xds/internal/client/lds_test.go

```diff
@@ -919,22 +1164,30 @@ func (s) TestUnmarshalListener_ServerSide(t *testing.T) {
 						IdentityCertName:     "identityCertName",
 						RequireClientCert:    true,
 					},
+					Raw: listenerWithValidationContext,
 				},
 			},
+			wantMD: UpdateMetadata{
+				Status:  ServiceStatusACKed,
+				Version: testVersion,
+			},
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			gotUpdate, _, err := UnmarshalListener("", test.resources, nil)
+			gotUpdate, md, err := UnmarshalListener(testVersion, test.resources, nil)
 			if (err != nil) != (test.wantErr != "") {
-				t.Fatalf("UnmarshalListener(%v) = %v wantErr: %q", test.resources, err, test.wantErr)
+				t.Errorf("UnmarshalListener(), got err: %v, wantErr: %v", err, test.wantErr)
```

Comment:
Even if the resource is NACK'ed, we still want to check the content in the update and metadata.
- update contains the resource names, so we know what was in the response
- metadata contains the resource version, and errors

Reply:
Oh, actually, that's a different thing.
Maybe this should stay Fatal.

Reply:
This change wasn't intentional. But it seems to be OK to `Error` instead of `Fatal`. I'm going to keep it this way. Let me know if you think otherwise.

Reply:
I see that we are using `Error` instead of `Fatal` in `TestUnmarshalListener_ClientSide` as well. But I don't see how useful a diff of the `update` or the `metadata` is going to be when the `gotErr != wantErr`.

Reply:
I updated the error check to `Fatal`. Kept the following checks as `Error`. It could be useful to compare metadata even if returned update is different.